packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map.ts

```diff
@@ -0,0 +1,770 @@
+   */
+  readonly csvHeaders?: CsvHeaders;
+}
+
+/**
+ * Item Reader configuration for iterating over items in a CSV file stored in S3
+ */
+export class S3CsvItemReader extends S3ItemReader {
+  /**
+   * CSV headers configuration
+   */
+  readonly csvHeaders: CsvHeaders;
+  protected readonly inputType: string = 'CSV';
+
+  constructor(props: S3CsvItemReaderProps) {
+    super(props);
+    this.csvHeaders = props.csvHeaders || CsvHeaders.useFirstRow();
+  }
+
+  public render(): any {
+    let rendered = super.render();
+
+    rendered.ReaderConfig = FieldUtils.renderObject({
+      ...rendered.ReaderConfig,
+      ...{
+        CSVHeaderLocation: this.csvHeaders.headerLocation,
+        ...(this.csvHeaders.headers && { CSVHeaders: this.csvHeaders.headers }),
+      },
+    });
+
+    return rendered;
+  }
+}
+
+/**
+ * Item Reader configuration for iterating over items in a S3 inventory manifest file stored in S3
+ */
+export class S3ManifestItemReader extends S3ItemReader {
+  protected readonly inputType: string = 'MANIFEST';
+}
+
+/**
+ * Interface for Result Writer configuration properties
+ */
+export interface ResultWriterProps {
+  /**
+   * S3 Bucket in which to save Map Run results
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * S3 prefix in which to save Map Run results
+   *
+   * @default - No prefix
+   */
+  readonly prefix?: string;
+}
+
+/**
+ * Configuration for writing Distributed Map state results to S3
+ */
+export class ResultWriter {
+  /**
+   * S3 Bucket in which to save Map Run results
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * S3 prefix in which to save Map Run results
+   *
+   * @default - No prefix
+   */
+  readonly prefix?: string;
+
+  constructor(props: ResultWriterProps) {
+    this.bucket = props.bucket;
+    this.prefix = props.prefix;
+  }
+
+  /**
+   * Render ResultWriter in ASL JSON format
+   */
+  public render(): any {
+    return FieldUtils.renderObject({
+      Resource: `arn:${Aws.PARTITION}:states:::s3:putObject`,
+      Parameters: {
+        Bucket: this.bucket.bucketName,
+        ...(this.prefix && { Prefix: this.prefix }),
+      },
+    });
+  }
+
+  /**
+   * Compile policy statements to provide relevent permissions to the state machine
+   */
+  public providePolicyStatements(): iam.PolicyStatement[] {
+
+    const resource = `arn:${Aws.PARTITION}:s3:::${this.bucket.bucketName}/*`;
+
+    return [
+      new iam.PolicyStatement({
+        actions: [
+          's3:PutObject',
+          's3:GetObject',
+          's3:ListMultipartUploadParts',
+          's3:AbortMultipartUpload',
+        ],
+        resources: [resource],
+      }),
+    ];
+  }
+}
+
+/**
+ * Interface for ItemBatcher configuration properties
+ */
+export interface ItemBatcherProps {
+  /**
+   * MaxItemsPerBatch
+   *
+   * Specifies the maximum number of items that each child workflow execution processes, as static number
+   *
+   * @default - No maxItemsPerBatch
```

Comment:
this does not tell me the actual default. same for the rest of the defaults here